rest/security-rest-cxfwrapper/src/test/java/org/codice/ddf/cxf/SecureCxfClientFactoryTest.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package org.codice.ddf.cxf;
+
+import ddf.security.Subject;
+import ddf.security.service.SecurityServiceException;
+import ddf.security.settings.SecuritySettingsService;
+import org.apache.commons.lang.StringUtils;
+import org.apache.cxf.configuration.jsse.TLSClientParameters;
+import org.apache.cxf.configuration.security.AuthorizationPolicy;
+import org.apache.cxf.jaxrs.client.Client;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.transport.http.HTTPConduit;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.session.mgt.SimpleSession;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.subject.support.DelegatingSubject;
+import org.junit.Test;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.MessageBodyReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SecureCxfClientFactoryTest {
```

Comment:
I'm guessing this will replace a good chunk of the DDF Spatial TrustedRemoteSource? Might be able to reuse some of the test code from in there: https://github.com/codice/ddf-spatial/blob/master/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java

Instead of creating a source instance in that example, you could just create a very basic (but real) rest client that actually hits an endpoint. This would be nice as you can test out the code fully instead of using mock objects for everything. I would assume that once this PR is pushed in and spatial is updated that the test I referenced would be eliminated.